File: Patterns/Manuscript/manuscript/code/java/DesignPatterns/src/main/java/bridge/RemoteControl.java
package bridge;

class RemoteControl {
    protected Device device;

    public RemoteControl(Device device) {
        this.device = device;
    }

    public void powerOn() {
        device.powerOn();
    }

    public void powerOff() {
        device.powerOff();
    }

    public void adjustVolume(int volume) {
        device.adjustVolume(volume);
    }

    public void changeChannel(int channel) {
        device.changeChannel(channel);
    }
}

File: code/csharp/DesignPatterns/bridge/RemoteControl.cs
namespace Bridge;

abstract class RemoteControl
{
    protected IDevice _device;

    public RemoteControl(IDevice device)
    {
        this._device = device;
    }

    public void PowerOn()
    {
        _device.PowerOn();
    }

    public void PowerOff()
    {
        _device.PowerOff();
    }

    public void AdjustVolume(int volume)
    {
        _device.AdjustVolume(volume);
    }

    public void ChangeChannel(int channel)
    {
        _device.ChangeChannel(channel);
    }
}
